ntoskrnl/se/sid.c

```diff
@@ -648,7 +733,13 @@ SeCaptureSidAndAttributesArray(
         }
 
         /* Set returned address to NULL */
-        *CapturedSidAndAttributes = NULL ;
+        *CapturedSidAndAttributes = NULL;
+    }
+
+    /* Free the temporary validation array */
+    if (PreviousMode != KernelMode)
```

Comment:
Add null check, in case the allocation or anything before it raises an exception.